src/main/java/de/unistuttgart/crosswordbackend/crosswordchecker/CrosswordGenerator.java

```diff
@@ -0,0 +1,261 @@
+package de.unistuttgart.crosswordbackend.crosswordchecker;
+
+import de.unistuttgart.crosswordbackend.data.Crossword;
+import de.unistuttgart.crosswordbackend.data.Intersection;
+import de.unistuttgart.crosswordbackend.data.Question;
+import java.security.SecureRandom;
+import java.util.*;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+public class CrosswordGenerator {
+
+  /**
+   * Count of crosswords generated and compared by getScore() function
+   */
+  private int interations = 10;
+  private int startRows = 10;
+  private int startColumns = 10;
+
+  private final String empty = "empty";
+
+  private SecureRandom random = new SecureRandom();
+
+  /**
+   * Count of tries of the algorithm to place a word.
+   */
+  private int maxTries = 100;
+
+  /**
+   * Genereates a crossword with the given questions
+   * @param questions Questions to create a crossword-puzzle with
+   * @return a Crossword-puzzle
+   */
+  public Crossword generateCrossword(List<Question> questions) {
+    Crossword crossword = createCrossword(questions);
+    for (int i = 0; i < interations; i++) {
+      Crossword currentCrossword = createCrossword(questions);
+      int crosswordScore = getScore(crossword);
+      int currentCrosswordScore = getScore(currentCrossword);
+      if (currentCrosswordScore > crosswordScore && currentCrossword.getAnswer().isEmpty()) {
+        crossword = currentCrossword;
+      }
+    }
```

Comment:
What are you doing here??

Reply:
Because the algorithm is a random algorithm, there may be the case that a crossword can be created, but because of the random order, in which the words get placed on the crossword, it wasn't possible in the current iteration. To combat this, the algorithm creates more crosswords and checks if there is one iteration of the generateCrossword algorithm where all words are placed on the crossword